common-arcana.lic

```diff
@@ -25,28 +25,32 @@ module DRCA
     end
   end
 
-  def activate_khri?(kneel, preps, ability)
+  def activate_khri?(kneel, ability)
     return false if DRSpells.active_spells[ability]
     DRCT.retreat if kneel
     DRC.bput('kneel', 'You kneel', 'You are already', 'You rise') if kneel && !kneeling?
-    result = DRC.bput(ability, preps)
+    result = DRC.bput(ability, get_data('spells').khri_preps)
     waitrt?
     DRC.fix_standing
 
     ['Your body is willing', 'You have not recovered'].include?(result)
   end
 
+  def prepare(abbrev, mana, command = 'prepare')
+    DRC.bput("#{command} #{abbrev} #{mana}", get_data('spells').prep_messages)
+  end
+
   def buff(spell, settings)
-    DRC.bput("pre #{spell['abbrev']} #{spell['mana']}", settings.prep_messages)
+    prepare(spell['abbrev'], spell['mana'])
     if spell['cambrinth']
       find_cambrinth(settings)
 
-      charge_and_invoke(settings, spell['cambrinth'])
+      charge_and_invoke(settings.cambrinth, spell['cambrinth'])
 
       stow_cambrinth(settings)
     end
     waitcastrt?
-    DRC.bput('cast', settings.cast_messages)
+    DRC.bput('cast', get_data('spells').cast_messages)
   end
 
   def find_cambrinth(settings)
```

Comment:
This would be my next target (i.e. to remove settings and pass in real args).

It occurred to me that if I invert the order of the conditionals (i.e. check stored first, then held), this method doesn't even need `held_cambrinth`... is that too risky?